packages/dds/tree/src/simple-tree/core/treeNodeSchema.ts

```diff
@@ -188,6 +226,11 @@ export interface TreeNodeSchemaCore<
 	 * @system
 	 */
 	readonly childTypes: ReadonlySet<TreeNodeSchema>;
+
+	/**
+	 * Gets the user-provided {@link NodeSchemaMetadata} for this node.
+	 */
+	readonly metadata: NodeSchemaMetadata<TCustomMetadata> | undefined;
```

Comment:
I think this API would be easier to use if we disallowed undefined and set this unconditionally. Since all the members are optional, we can construct a value for it in all cases.
```suggestion
	readonly metadata: NodeSchemaMetadata<TCustomMetadata>;
```